### PR TITLE
Homework 4

### PR DESCRIPTION
**Pull request by slavrikk**

### CODE REVIEW

File: .idea/workspace.xml [hunk 1]
```diff
@@ -21,6 +21,7 @@
   <component name="ChangeListManager">
```
Comment (shoggy) [inline .idea/workspace.xml]: Remove this file from git index please
```sh
git rm --cached .idea/workspace.xml
```

File: pom.xml [hunk 2]
```diff
@@ -9,26 +9,37 @@
...
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-surefire-plugin</artifactId>
                 <version>${maven-surefire-version}</version>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <source>8</source>
```
Comment (shoggy) [inline pom.xml]: The compiler version is configured in `properties` section and here.
Please choose only one, to avoid ambiguities

File: pom.xml [hunk 3]
```diff
@@ -9,26 +9,37 @@
...
+                    <target>8</target>
+                </configuration>
+            </plugin>
         </plugins>
     </build>
 
+
     <dependencies>
 
         <dependency>
```
Comment (shoggy) [inline pom.xml:51]: I think this declaration is redundant because this is plugin and it is already defined in `plugins`

File: pom.xml [hunk 4]
```diff
@@ -112,5 +150,126 @@
...
+                            <reportVersion>${allure.version}</reportVersion>
+                            <resultsDirectory>${project.build.directory}/allure-results</resultsDirectory>
+                        </configuration>
+                        <dependencies>
+                            <!-- https://mvnrepository.com/artifact/io.qameta.allure/allure-maven -->
+                            <dependency>
+                                <groupId>io.qameta.allure</groupId>
+                                <artifactId>allure-maven</artifactId>
+                                <version>2.10.0</version>
+                            </dependency>
```
Comment (shoggy) [inline pom.xml]: What for the plugin has itself in dependencies?

File: pom.xml [hunk 5]
```diff
@@ -112,5 +150,126 @@
...
+                        </dependencies>
+                    </plugin>
+                    <plugin>
+                        <groupId>io.qameta.allure</groupId>
+                        <artifactId>allure-maven</artifactId>
+                        <version>2.10.0</version>
+                        <configuration>
+                            <reportVersion>${allure.version}</reportVersion>
+                            <resultsDirectory>${project.build.directory}/allure-results</resultsDirectory>
```
Comment (shoggy) [inline pom.xml]: This property is defined here and in `allure.properties`. Choose one place, please

**Review by shoggy — CHANGES_REQUESTED** [review]
`allure-results` and `.idea` in a repository make me sad

`@FindBy` looks not so good -- please merge in changes after Konstantin approve hw2/3

File: .idea/workspace.xml (hunk 1 above)
Comment (slavrikk) [reply .idea/workspace.xml]: done

File: pom.xml (hunk 3 above)
Comment (slavrikk) [reply pom.xml:51]: done

File: pom.xml (hunk 5 above)
Comment (slavrikk) [reply pom.xml]: done

File: pom.xml (hunk 2 above)
Comment (slavrikk) [reply pom.xml]: fix

File: pom.xml (hunk 4 above)
Comment (slavrikk) [reply pom.xml]: remove

**Review by shoggy — APPROVED** [review]
Looks good!

Note: job config.xml sent personally

**Pull request merged**